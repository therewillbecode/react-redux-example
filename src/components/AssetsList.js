import React from "react";
import { Link } from "react-router-dom";
import { Table } from "semantic-ui-react";

const AssetsList = ({ assets, navigateToAsset }) => (
  <Table celled padded inverted selectable>
    <Table.Header>
      <Table.Row>
        <Table.HeaderCell textAlign="center">ID</Table.HeaderCell>
        <Table.HeaderCell textAlign="center">Name</Table.HeaderCell>
        <Table.HeaderCell textAlign="center">Status</Table.HeaderCell>
        <Table.HeaderCell textAlign="center">Time Received</Table.HeaderCell>
        <Table.HeaderCell textAlign="center">Time Dispatched</Table.HeaderCell>
        <Table.HeaderCell textAlign="center">Comments</Table.HeaderCell>
      </Table.Row>
    </Table.Header>

    <Table.Body>
      {assets.map(
        ({
          id,
          name,
          status,
          receivedTimestamp,
          dispatchedTimestamp,
          comment
        }) => (
          <Table.Row key={id} onClick={() => navigateToAsset(id)}>
            <Table.Cell textAlign="center">{id}</Table.Cell>
            <Table.Cell textAlign="center">{name}</Table.Cell>
            <Table.Cell textAlign="center">{status}</Table.Cell>
            <Table.Cell textAlign="center">{receivedTimestamp}</Table.Cell>
            <Table.Cell textAlign="center">{dispatchedTimestamp}</Table.Cell>
            <Table.Cell textAlign="left">{comment}</Table.Cell>
          </Table.Row>
        )
      )}
    </Table.Body>
  </Table>
);

export default AssetsList;
