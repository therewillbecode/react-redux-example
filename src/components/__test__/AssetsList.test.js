/* eslint-disable */
import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";

import AssetsList from "../AssetsList";

const assets = [
  {
    id: "kj48",
    name: "pfs",
    status: "dispatched",
    receivedTimestamp: "34245",
    dispatchedTimestamp: "5235",
    comment: "Lorem"
  }
];

describe("AssetsList", () => {
  it("renders correctly", () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<AssetsList assets={assets} />);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
