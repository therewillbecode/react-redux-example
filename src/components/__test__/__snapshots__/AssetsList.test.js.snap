// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetsList renders correctly 1`] = `
<Table
  as="table"
  celled={true}
  inverted={true}
  padded={true}
  selectable={true}
>
  <TableHeader
    as="thead"
  >
    <TableRow
      as="tr"
      cellAs="td"
    >
      <TableHeaderCell
        as="th"
        textAlign="center"
      >
        <h3>
          ID
        </h3>
      </TableHeaderCell>
      <TableHeaderCell
        as="th"
        textAlign="center"
      >
        <h3>
          Name
        </h3>
      </TableHeaderCell>
      <TableHeaderCell
        as="th"
        textAlign="center"
      >
        <h3>
          Status
        </h3>
      </TableHeaderCell>
      <TableHeaderCell
        as="th"
        textAlign="center"
      >
        <h3>
          Time Received
        </h3>
      </TableHeaderCell>
      <TableHeaderCell
        as="th"
        textAlign="center"
      >
        <h3>
          Time Dispatched
        </h3>
      </TableHeaderCell>
      <TableHeaderCell
        as="th"
        textAlign="center"
      >
        <h3>
          Comments
        </h3>
      </TableHeaderCell>
    </TableRow>
  </TableHeader>
  <TableBody
    as="tbody"
  >
    <TableRow
      as="tr"
      cellAs="td"
      onClick={[Function]}
    >
      <TableCell
        as="td"
        textAlign="center"
      >
        <h5>
          kj48
        </h5>
      </TableCell>
      <TableCell
        as="td"
        textAlign="center"
      >
        <h5>
          pfs
        </h5>
      </TableCell>
      <TableCell
        as="td"
        textAlign="center"
      >
        <h5>
          dispatched
        </h5>
      </TableCell>
      <TableCell
        as="td"
        textAlign="center"
      >
        <h5>
          34245
        </h5>
      </TableCell>
      <TableCell
        as="td"
        textAlign="center"
      >
        <h5>
          5235
        </h5>
      </TableCell>
      <TableCell
        as="td"
        textAlign="center"
      >
        <h5>
          Lorem
        </h5>
      </TableCell>
    </TableRow>
  </TableBody>
</Table>
`;
